securedrop/journalist.py

```diff
@@ -120,38 +120,50 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def validate_user(username, password, token):
+    """
+    Validates the user by calling the login and handling exceptions
+    :param username: Username
+    :param password: Password
+    :param token: Two-factor authentication token
+    :return: Journalist user object if successful, None otherwise.
+    """
+    try:
+        return Journalist.login(username, password, token)
+    except Exception as e:
+        app.logger.error("Login for '{}' failed: {}".format(
+            username, e))
+        login_flashed_msg = gettext('Login failed.')
```

Comment:
In the context of a password change, this error message should be different. Something like **Incorrect password or two-factor code** maybe ?

Reply:
I did notice the error wasn't exactly the best. I'm wondering if "Authentication failed" would be better to support both? The other parts about waiting or using a different 2FA code refer to logging in, so I'd like to try to keep it generic (and hopefully help reduce the amount of text that needs to be translated).

I'll change this depending on context (basically passing in the string to the function), and change the rest of the notices to be generic with something like "before trying again" (instead of "before logging in again").